### PR TITLE
Add ability to use secrets for all mysql access params

### PR DESCRIPTION
**Pull request by skimpax**
Currently, only password is got from docker secrets.
This PR allow to set all parameters (user, password, database name), as is is done in the mariadb official docker image.

### CODE REVIEW

File: backup.sh [hunk 1]
```diff
@@ -1,10 +1,18 @@
 #!/bin/bash
 
+# Get hostname: try read from file, else get from env
+[ -z "${MYSQL_HOST_FILE}" ] || { MYSQL_USER=$(head -1 "${MYSQL_HOST_FILE}"); }
```
Comment (fradelg) [inline backup.sh:4]: I think you want to extract the hostname from the file, not the MYSQL_USER. Check this assignment

File: restore.sh [hunk 2]
```diff
@@ -1,10 +1,14 @@
 #!/bin/bash
 
+# Get hostname: try read from file, else get from env
+[ -z "${MYSQL_HOST_FILE}" ] || { MYSQL_USER=$(head -1 "${MYSQL_HOST_FILE}"); }
```
Comment (fradelg) [inline restore.sh:4]: This is the same issue you have with backup.sh where HOST and USER are mixed in the same line

File: README.md [hunk 3]
```diff
@@ -18,9 +18,11 @@ docker container run -d \
 - `MYSQL_HOST`: The host/ip of your mysql database.
```
Comment (fradelg) [inline README.md:18]: Shouldn't you add here the MYSQL_HOST_FILE too?

File: README.md [hunk 4]
```diff
@@ -18,9 +18,11 @@ docker container run -d \
...
 - `MYSQL_PORT`: The port number of your mysql database.
 - `MYSQL_USER`: The username of your mysql database.
+- `MYSQL_USER_FILE`: The file in container where to find the user of your mysql database (cf. docker secrets). You should use either MYSQL_USER_FILE or MYSQL_USER (see examples below).
 - `MYSQL_PASS`: The password of your mysql database.
 - `MYSQL_PASS_FILE`: The file in container where to find the password of your mysql database (cf. docker secrets). You should use either MYSQL_PASS_FILE or MYSQL_PASS (see examples below).
 - `MYSQL_DATABASE`: The database name to dump. Default: `--all-databases`.
+- `MYSQL_DATABASE_FILE`: The file in container where to find the database name in your mysql database (cf. docker secrets). You should use either MYSQL_DATABASE or MYSQL_DATABASE_FILE (see examples below).
```
Comment (fradelg) [inline README.md:25]: Please, add more details about the contents of this file (one single database name per line) or provide a very brief example

**Comment by skimpax** [comment]
Maybe I woke up too early, still closed eyes... never did a so crappy PR before.
Hoping these new commits fix all.

**Comment by fradelg** [comment]
hehe, no worries, thank you for your contribution!

**Pull request merged**